### PR TITLE
enable strict mode in select package

### PR DESCRIPTION
**Pull request by giladgray**
also add `| null` type to all ref props in all packages (for compatibility)
and fix a React warning in Popover example

### CODE REVIEW

**Comment by giladgray** [comment]
this is blocked on #2209 to resolve the constructor issues 😄

File: packages/select/src/components/select/multiSelect.tsx [hunk 1]
```diff
@@ -235,7 +233,7 @@ export class MultiSelect<T> extends React.PureComponent<IMultiSelectProps<T>, IM
             if (which === Keys.ESCAPE || which === Keys.TAB) {
                 // By default the escape key will not trigger a blur on the
                 // input element. It must be done explicitly.
-                this.input.blur();
+                this.input && this.input.blur();
```
Comment (adidahiya) [inline packages/select/src/components/select/multiSelect.tsx]: please don't do this type coercion for statements; it's confusing. we can be more lenient about inline expressions (but I personally still don't love it)

File: packages/select/src/components/select/suggest.tsx [hunk 2]
```diff
@@ -77,11 +77,9 @@ export class Suggest<T> extends React.PureComponent<ISuggestProps<T>, ISuggestSt
     // one, necessarily non-`static` place. note that Partial is necessary to
     // avoid having to define all required props, and all of these values must
     // be referenced manually in code.
-    private DEFAULT_PROPS: Partial<ISuggestProps<T>> = {
+    private DEFAULT_PROPS = {
```
Comment (adidahiya) [inline packages/select/src/components/select/suggest.tsx]: why remove the typedef?

Comment (adidahiya) [reply packages/select/src/components/select/suggest.tsx]: also I think you can make this `static`

Comment (giladgray) [reply packages/select/src/components/select/suggest.tsx]: the typedef preserves the `| undefined` from optional props, which doesn't help anyone

**Pull request merged**